ddtrace/opentracer/propagation/http.py

```diff
@@ -0,0 +1,85 @@
+import logging
+
+from opentracing import InvalidCarrierException, SpanContextCorruptedException
+from ddtrace.propagation.http import HTTPPropagator as DDHTTPPropagator
+from ddtrace.propagation.http import HTTP_HEADER_TRACE_ID, HTTP_HEADER_PARENT_ID, HTTP_HEADER_SAMPLING_PRIORITY
+
+from ..span_context import SpanContext
+
+from .propagator import Propagator
+
+
+log = logging.getLogger(__name__)
+
+HTTP_BAGGAGE_PREFIX = 'ot-baggage-'
+HTTP_BAGGAGE_PREFIX_LEN = len(HTTP_BAGGAGE_PREFIX)
+
+# valid baggage keys to check span validity
+VALID_BAGGAGE_KEYS = {
+    HTTP_HEADER_TRACE_ID: True,
+    HTTP_HEADER_PARENT_ID: True,
+    HTTP_HEADER_SAMPLING_PRIORITY: True,
+}
+
+
+class HTTPPropagator(Propagator):
+    """OpenTracing compatible HTTP_HEADER and TEXT_MAP format propagator.
+
+    `HTTPPropagator` provides compatibility by using existing OpenTracing
+    compatible methods from the ddtracer along with new logic supporting the
+    outstanding OpenTracing-defined functionality.
+    """
+
+    __slots__ = ['_dd_propagator']
+
+    def __init__(self):
+        self._dd_propagator = DDHTTPPropagator()
+
+    def inject(self, span_context, carrier):
+        """Inject a span context into a carrier.
+
+        *span_context* is injected into the carrier by first using an
+        :class:`ddtrace.propagation.http.HTTPPropagator` to inject the ddtracer
+        specific fields.
+
+        Then the baggage is injected into *carrier*.
+
+        :param span_context: span context to inject.
+
+        :param carrier: carrier to inject into.
+        """
+        if not isinstance(carrier, dict):
+            raise InvalidCarrierException('propagator expects carrier to be a dict')
+
+        self._dd_propagator.inject(span_context._context, carrier)
+
+        # Add the baggage
+        if span_context.baggage is not None:
+            for key in span_context.baggage:
+                carrier[HTTP_BAGGAGE_PREFIX + key] = span_context.baggage[key]
+
+    def extract(self, carrier):
+        """Extract a span context from a carrier.
+
+        :class:`ddtrace.propagation.http.HTTPPropagator` is used to extract
+        ddtracer supported fields into a `ddtrace.Context` context which is
+        combined with new logic to extract the baggage which is returned in an
+        OpenTracing compatible span context.
+
+        :param carrier: carrier to extract from.
+
+        :return: extracted span context.
+        """
+        if not isinstance(carrier, dict):
+            raise InvalidCarrierException('propagator expects carrier to be a dict')
+
+        ddspan_ctx = self._dd_propagator.extract(carrier)
+
+        baggage = {}
+        for key in carrier:
+            if key.startswith(HTTP_BAGGAGE_PREFIX):
+                baggage[key[HTTP_BAGGAGE_PREFIX_LEN:]] = carrier[key]
+            elif key not in VALID_BAGGAGE_KEYS:
+                raise SpanContextCorruptedException('invalid key in span context')
```

Comment:
Do we need that? I guess a `carrier` could be an HTTP header. So it has `HTTP_HEADER_TRACE_ID` and other things. Some of them are related to the baggage, but if users are propagating custom things such as `HTTP_LOG_DEBUG_ID`? We're raising an exception all the time making impossible the extraction.

I think we should extract the span context with our internal propagator and then retrieve the baggage. Any other header or field must simply be ignored. Am I missing a use case?

Reply:
No you're right. I overlooked with this check that the carrier may contain keys that are user defined. I'll remove the check.